src/os/FileJournal.cc

```diff
@@ -1666,6 +1675,20 @@ void FileJournal::pop_write()
   writeq.pop_front();
 }
 
+void FileJournal::batch_peek_write(list<write_item> &items)
+{
+  assert(write_lock.is_locked());
+  Mutex::Locker locker(writeq_lock);
+  writeq.swap(items);
+}
+
+void FileJournal::backfill_write(list<write_item> &items)
+{
+  assert(write_lock.is_locked());
+  Mutex::Locker locker(writeq_lock);
+  writeq.splice(writeq.begin(), items);
+}
+
```

Comment:
I'd call these batch_shift_writeq() and batch_unshift_writeq()?  (shift/unshift being the opposite of pop/push, as in perl).